packages/cli/src/lib/axe-test-urls.ts

```diff
@@ -91,7 +92,8 @@ const testPages = async (
 
           /* istanbul ignore if */
           if (err) {
-            return reject(err);
+            console.error(error(err));
+            process.exit(1);
```

Comment:
Why doesn't `reject(err)` allow the caller to handle the error on their own? I don't understand how/why this fixes anything.

Reply:
`reject(err)` doesn't provide a useful error message to the user: 

![image](https://user-images.githubusercontent.com/41127686/148828375-bbce5da4-94b3-4a03-b8f0-a7b632e227b8.png)

Including `--show-errors` flag does not either: 

![image](https://user-images.githubusercontent.com/41127686/148828464-43c6bfa5-cf0b-4839-9f7d-fac89c6b45d2.png)

Reply:
That seems like the problem is how we're handing rejections then.

Reply:
It looks like the issue is here:

https://github.com/dequelabs/axe-core-npm/blob/v4.3.2/packages/cli/src/bin/index.ts#L153

What is `%j \n $s` supposed to do?

Reply:
I think the error is here https://github.com/dequelabs/axe-core-npm/blob/v4.3.2/packages/cli/src/lib/axe-test-urls.ts#L87 we use `err: string` so we don't have access to `e.message` or `e.stack` to do the the string formatting you mentioned

Reply:
I see. Yes, we should _only_ be rejecting with `Error`s, not other things. Rejecting/`throw`ing anything but an instance of an `Error` is considered a bad practice and should be avoided. [ESLint protects us from this sort of thing with `throw`](https://eslint.org/docs/rules/no-throw-literal), but doesn't do the same for Promise rejections. We should pass the entire error back, not just a string.

I didn't know about `%j`, thanks for that. FWIW the `Error` class does not have a `.toJSON()`, so I'm pretty sure that's not doing anything.

Reply:
If the types on the line you linked are correct (`err: string`), then we should do something like:

```ts
        axe.analyze((err: Error | string | null, results: AxeResults) => {
          if (config.timer) {
            events?.endTimer('axe-core execution time');
          }

          /* istanbul ignore if */
          if (err) {
            const error = typeof err === 'string' ? new Error(err) : err
            return reject(error);
          }
```

Reply:
It looks like that type is wrong, as we [pass back the whole `Error` (unless in legacy/callback mode)](https://github.com/dequelabs/axe-core-npm/blob/v4.3.2/packages/webdriverjs/src/index.ts#L134).

Reply:
Ya, looking at it a bit more, the [type for the callback function](https://github.com/dequelabs/axe-core-npm/blob/v4.3.2/packages/webdriverjs/src/types.ts#L20) should be `Error | null` and not `string | null`. So we can access `e.message` etc. However, your work around does allow the user to _handle_ the error with `--show-errors`

![image](https://user-images.githubusercontent.com/41127686/148837170-fc1ab1fa-8a13-4a9f-a312-648f3f13f4c7.png)

Reply:
Yes, the type for that function is wrong. IMO it should be changed/corrected.

However, if we want a _very simple fix_ for the issue at hand, we could just fix the `console.error()` call in `src/bin/index.ts`:

```js
    if (!showErrors) {
      console.error(error('An error occurred while testing this page.'));
    } else {
      console.error(error('Error: %s'), e);
    }
```

Reply:
For reference issue has been raised: https://github.com/dequelabs/axe-core-npm/issues/422